supabase/functions/stats/index.ts

```diff
@@ -13,9 +14,54 @@ const failActions = [
   'update_fail',
   'download_fail',
 ]
+
+// ios sends 13 fields while android sends 11 fields
+const jsonRequestSchema = z.object({
+  app_id: z.string({
+    required_error: 'App ID is required',
+    invalid_type_error: 'App ID name must be a string',
+  }),
+  device_id: z.string({
+    required_error: 'Device ID is required',
+    invalid_type_error: 'Device ID must be a string',
+  }).max(36),
+  platform: z.string({
+    required_error: 'Platform type is required',
+    invalid_type_error: 'Platform type must be a string',
+  }),
+  version_name: z.string({
+    required_error: 'Version name is required',
+    invalid_type_error: 'Version name must be a string',
+  }),
+  version_os: z.string({
+    required_error: 'Version OS is required',
+    invalid_type_error: 'Version OS must be a string',
+  }),
+  version_code: z.optional(z.string()),
+  version_build: z.optional(z.string()),
+  action: z.optional(z.string()),
+  custom_id: z.optional(z.string()),
+  channel: z.optional(z.string()),
+  plugin_version: z.optional(z.string()),
+  is_emulator: z.boolean().default(false),
+  is_prod: z.boolean().default(true),
+}).refine(data => reverseDomainRegex.test(data.app_id), {
+  message: 'App ID name must be a reverse domain string',
+}).refine(data => deviceIdRegex.test(data.device_id), {
+  message: 'Device ID must be a valid UUID string',
+})
+
 async function main(url: URL, headers: BaseHeaders, method: string, body: AppStats) {
   try {
     console.log('body', body)
+    let _parseResult = null
```

Comment:
Why is this unused? The entire idea is to add type safety and if you do not use zod result you do not have type safety

Reply:
Fixed in the latest [commit 89a985](https://github.com/Cap-go/capgo/commit/89a985b8f1ec926269aed3fa5f7134b7f8afca6e).